tests/unit/game/effects/weakness.test.ts

```diff
@@ -0,0 +1,321 @@
+				},
+			},
+			{startWithAllCards: true, noItemRequirements: true},
+		)
+	})
+
+	test('Weakness Self Redundancy', () => {
+		testGame(
+			{
+				playerOneDeck: [VintageBeefCommon],
+				playerTwoDeck: [VintageBeefCommon, PotionOfWeakness, PotionOfWeakness],
+				saga: function* (game) {
+					yield* playCardFromHand(game, VintageBeefCommon, 'hermit', 0)
+					yield* endTurn(game)
+
+					yield* playCardFromHand(game, VintageBeefCommon, 'hermit', 0)
+					yield* playCardFromHand(game, PotionOfWeakness, 'single_use')
+					yield* applyEffect(game)
+					yield* endTurn(game)
+
+					yield* endTurn(game)
+
+					yield* playCardFromHand(game, PotionOfWeakness, 'single_use')
+					yield* applyEffect(game)
+					yield* attack(game, 'primary')
+					yield* endTurn(game)
+
+					expect(
+						game.components.find(
+							RowComponent,
+							query.row.currentPlayer,
+							query.row.index(0),
+						)?.health,
+					).toBe(
+						VintageBeefCommon.health - VintageBeefCommon.primary.damage - 20,
+					)
+				},
+			},
+			{startWithAllCards: true, noItemRequirements: true},
+		)
+	})
+
+	test('Weakness Heteroredundancy', () => {
+		testGame(
+			{
+				playerOneDeck: [SmallishbeansRare],
+				playerTwoDeck: [ImpulseSVCommon, PotionOfWeakness],
+				saga: function* (game) {
+					yield* playCardFromHand(game, SmallishbeansRare, 'hermit', 0)
+					yield* endTurn(game)
+
+					yield* playCardFromHand(game, ImpulseSVCommon, 'hermit', 0)
+					yield* playCardFromHand(game, PotionOfWeakness, 'single_use')
+					yield* applyEffect(game)
+					yield* attack(game, 'primary')
+					yield* endTurn(game)
+
+					expect(
+						game.components.find(
+							RowComponent,
+							query.row.currentPlayer,
+							query.row.index(0),
+						)?.health,
+					).toBe(SmallishbeansRare.health - ImpulseSVCommon.primary.damage - 20)
+				},
+			},
+			{startWithAllCards: true, noItemRequirements: true},
+		)
+	})
+
+	test('Weakness Lethality Check', () => {
+		testGame(
+			{
+				playerOneDeck: [ImpulseSVCommon, SmallishbeansRare, ImpulseSVCommon],
+				playerTwoDeck: [
+					SmallishbeansRare,
+					SmallishbeansRare,
+					SmallishbeansRare,
+					PotionOfWeakness,
+				],
+				saga: function* (game) {
+					yield* playCardFromHand(game, ImpulseSVCommon, 'hermit', 0)
+					yield* playCardFromHand(game, SmallishbeansRare, 'hermit', 1)
+					yield* playCardFromHand(game, ImpulseSVCommon, 'hermit', 2)
+					yield* endTurn(game)
+
+					yield* playCardFromHand(game, SmallishbeansRare, 'hermit', 0)
+					yield* playCardFromHand(game, SmallishbeansRare, 'hermit', 1)
+					yield* playCardFromHand(game, SmallishbeansRare, 'hermit', 2)
+					yield* attack(game, 'secondary')
+					yield* endTurn(game)
+
+					yield* endTurn(game)
+
+					yield* playCardFromHand(game, PotionOfWeakness, 'single_use')
+					yield* applyEffect(game)
+					yield* attack(game, 'secondary')
+					yield* endTurn(game)
+
+					yield* changeActiveHermit(game, 1)
```

Comment:
Since this is supposed to check Weakness still works after a knock-out, you need one more AFK Joel to deal enough damage. Add this line to verify a knock-out did occur
```suggestion
					expect(game.currentPlayer.activeRow).toBe(null)
					yield* changeActiveHermit(game, 1)
```

Reply:
That damage was covered by weakness. Though that's definitely a useful addition.

Reply:
Or apply weakness the first time you use the secondary since it is still 10hp short of a KO

Reply:
I forgot I used Impy for that part and my brain switched back to Joel for the rest of the construction lol.